gwpy/types/series.py

```diff
@@ -769,37 +769,43 @@ def append(self, other, inplace=True, pad=None, gap=None, resize=True):
         if pad is None and gap == 'pad':
             pad = 0.
 
-        # check metadata
-        self.is_compatible(other)
+        # check metadata (do not do this if trying to join to unevenly sampled
+        # discontiguous series')
+        if not (
+            gap == 'ignore' and (
+                not hasattr(self, 'dx') or not hasattr(other, 'dx')
+            )
+        ):
```

Comment:
@mattpitkin, is this a simpler version of what you are suggesting:

```suggestion
        if (
            gap != "ignore"
            or (hasattr(self, "dx") and hasattr(other, "dx"))
        ):
```